src/_guides/language/common-prob.md

```diff
@@ -0,0 +1,473 @@
+</div>
+
+As a last resort, you can cast the type using <code>as <em>Class</em></code>.
+This solution is risker because the cast silences the static error by
+inserting a runtime cast that may fail at runtime.
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+var stuff = []; // Runtime type is List<dynamic>.
+stuff.add("Hi");
+List<String> strings = stuff [[highlight]]as List<String>[[/highlight]];
+{% endprettify %}
+</div>
+
+Note: Currently, only DDC makes strong mode
+runtime checks, but it's coming to other tools.
+
+In more complex situations where this warning appears, you may want
+to use a generic method. You can either use an existing method, such
+as `Iterable.map()`, or you can define your own.
+For more information, see [Using generic methods](/guides/language/language-tour#using-generic-methods)
+in the [language tour](/guides/language/language-tour).
+
+<hr>
+
+<a name="missing-type-arguments"></a>
+### Missing type arguments
+
+Omitting a type argument when defining a generic subclass can cause one
+of two kinds of problems during static analysis:
+
+**Error:** <code>Invalid override. The type of &lt;<em>type</em>&gt; is not a subtype of &lt;<em>type</em>&gt;.</code>
+
+OR
+
+**Warning:** <code>Unsound implicit cast from <em>Class&lt;dynamic&gt;</em> to <em>Class&lt;type</em>&gt;.</code>
+
+**Fix:** Specify type arguments for the generic subclass.
+
+When a generic subclass neglects to specify a type argument,
+the analyzer infers the `dynamic` type. This is likely to cause
+errors like invalid overrides or unsound downcasts.
+
+In the following example, Subclass extends Superclass<T> but doesn't
+specify a type argument. The analyzer infers Subclass&lt;dynamic&gt;
+which results in an invalid override error on `method(int)`.
+
+<div class="fails-sa" markdown="1">
+{% prettify dart %}
+class Superclass<T> {
+  void method(T t) {}
+}
+
+class Subclass extends Superclass {
+  [[highlight]]void method(int i) {}[[/highlight]] // <-- Error.
+}
+{% endprettify %}
+</div>
+
+You can fix this by specifying the type on the subclass:
+
+<div class="passes-sa" markdown="1">
+{% prettify dart %}
+class Superclass<T> {
+  void method(T t) {}
+}
+
+class Subclass extends Superclass[[highlight]]<int>[[/highlight]] {
+  void method(int i) {}
+}
+{% endprettify %}
+</div>
+
+<hr>
+
+<a name="assigning-mismatched-types"></a>
+### Assigning mismatched types
+
+**Warning:** <code>A value of type '&lt;<em>type</em>&gt;' cannot be
+assigned to a variable of type '<em>type</em>'.</code>
+
+This sometimes happens when you create a simple dynamic collection
+and the analyzer Dart infers the type in a way you didn't expect.
+When you later add values of a different type,
+the analyzer produces a warning.
+
+**Fix:** Specify the type explicitly.
+
+For example, the following code initializes a map with several
+(Sting,integer) pairs. The analyzer infers that map to be of type
+<String,int> but the code assumes <String,dynamic>.
```

Comment:
I think all this `<TypeA,TypeB>` occurrences need to be in code font.

Should they have space after the comma? I thought so, but it's so consistently not that way...